cdisc_rules_engine/services/data_services/base_data_service.py

```diff
@@ -132,15 +132,31 @@ def concat_split_datasets(
             func_to_call, dataset_names, **kwargs
         )
 
-        # concat datasets
-        full_dataset: pd.DataFrame = pd.concat(
-            [dataset for dataset in datasets],
-            ignore_index=True,
-        )
+        full_dataset = pd.DataFrame()
+        for dataset in datasets:
+            if "RDOMAIN" in dataset.columns:
+                supp_merged_dataset = self.merge_supp_dataset(full_dataset, dataset)
+                full_dataset = supp_merged_dataset
+            else:
+                full_dataset = pd.concat([full_dataset, dataset], ignore_index=True)
+
         if drop_duplicates:
             full_dataset.drop_duplicates()
         return full_dataset
 
+    def merge_supp_dataset(self, full_dataset, supp_dataset):
+        merge_keys = ["STUDYID", "USUBJID", "APID", "POOLID", "SPDEVID"]
+        merged_df = pd.merge(
+            full_dataset,
+            supp_dataset,
+            how="inner",
+            on=merge_keys,
+            left_on="IDVAR",
+            right_on="IDVARVAL",
+        )
+        print(merged_df.head())
```

Comment:
did you mean to leave this print statement in or is this an artifact from debugging?

Reply:
They were for debugging. Removed.